romana/openstack/tenant.go

```diff
@@ -0,0 +1,238 @@
+			log.Println("Error: ", err)
+			return nil, err
+		}
+	}
+	return identityClient, nil
+}
+
+func getComputeClient() (*gophercloud.ServiceClient, error) {
+	var err error
+	if computeClient == nil {
+		computeClient, err = initComputeClient()
+		if err != nil {
+			log.Println("Error: ", err)
+			return nil, err
+		}
+	}
+	return computeClient, nil
+}
+
+func getNetworkClient() (*gophercloud.ServiceClient, error) {
+	var err error
+	if networkClient == nil {
+		networkClient, err = initNetworkClient()
+		if err != nil {
+			log.Println("Error: ", err)
+			return nil, err
+		}
+	}
+	return networkClient, nil
+}
+
+// initIdentityClient initializes openstack api using
+// gophercloud. Currently it uses enviornmental variables
+// for authenticating with openstack identity.
+func initIdentityClient() (*gophercloud.ServiceClient, error) {
+	opts, err := openstack.AuthOptionsFromEnv()
+	if err != nil {
+		log.Println("Error fetching openstack env vars: ", err)
+		return nil, err
+	}
+	provider, err := openstack.AuthenticatedClient(opts)
+	if err != nil {
+		log.Println("Error authenticating with openstack: ", err)
+		return nil, err
+	}
+	return openstack.NewIdentityV2(provider), nil
+}
+
+// initComputeClient initializes openstack api using
+// gophercloud. Currently it uses enviornmental variables
+// for authenticating with openstack identity.
+func initComputeClient() (*gophercloud.ServiceClient, error) {
+	opts, err := openstack.AuthOptionsFromEnv()
+	if err != nil {
+		log.Println("Error fetching openstack env vars: ", err)
+		return nil, err
+	}
+	provider, err := openstack.AuthenticatedClient(opts)
+	if err != nil {
+		log.Println("Error authenticating with openstack: ", err)
+		return nil, err
+	}
+	return openstack.NewComputeV2(provider, gophercloud.EndpointOpts{
+		Name:   "compute",
+		Region: os.Getenv("OS_REGION_NAME"),
+	})
+}
+
+// initNetworkClient initializes openstack api using
+// gophercloud. Currently it uses enviornmental variables
+// for authenticating with openstack identity.
+func initNetworkClient() (*gophercloud.ServiceClient, error) {
+	opts, err := openstack.AuthOptionsFromEnv()
+	if err != nil {
+		log.Println("Error fetching openstack env vars: ", err)
+		return nil, err
+	}
+	provider, err := openstack.AuthenticatedClient(opts)
+	if err != nil {
+		log.Println("Error authenticating with openstack: ", err)
+		return nil, err
+	}
+	return openstack.NewNetworkV2(provider, gophercloud.EndpointOpts{
+		Name:   "neutron",
+		Region: os.Getenv("OS_REGION_NAME"),
+	})
+}
+
+// GetTenantName returns openstack tenant name corresponding
+// to the UUID being used in romana tenants.
+func GetTenantName(uuid string) (string, error) {
+	var tenant string
+
+	c, err := getIdentityClient()
+	if err != nil {
+		log.Println("Error getting Identity Client: ", err)
+		return "", err
+	}
+
+	opts := tenants.ListOpts{Limit: 20}
+	pager := tenants.List(c, &opts)
+	// brute force the whole tenant list to get the name?
+	pager.EachPage(
```

Comment:
I think unpleasant code like this shows us that our REST interface on our services needs a simple query interface.